crypto/params.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/bn.h>
+#include <openssl/params.h>
+#include <openssl/err.h>
+#include <openssl/cryptoerr.h>
+
+OSSL_PARAM *OSSL_PARAM_locate(const OSSL_PARAM *p, const char *key)
+{
+    for (; p->key != NULL; p++)
+        if (strcmp(key, p->key) == 0)
+            return (OSSL_PARAM *)p;
```

Comment:
This shouldn't cast away the const. If it really needs to return a non-const then the input parameter shouldn't be const either.